JumpRoyale/src/TimerOverlay.cs

```diff
@@ -0,0 +1,62 @@
+using System.Threading.Tasks;
+using Godot;
+
+public partial class TimerOverlay : VFlowContainer
+{
+    private const string SpriteName = "Sprite2D";
+    private const string TimerNodeName = SpriteName + "/Timer";
+
+    private static readonly int GameLength = GameOverlay.GameLength;
+
+    private int _timerSeconds = GameLength;
+
+    [Signal]
+    public delegate void TimerDoneEventHandler();
+
+    public void Init()
+    {
+        _ = StartTimer();
+    }
+
+    public async Task StartTimer()
+    {
+        await ToSignal(GetTree().CreateTimer(1.0f), SceneTreeTimer.SignalName.Timeout);
+
+        UpdateTimer();
+
+        if (_timerSeconds <= 0)
+        {
+            EmitSignal(SignalName.TimerDone);
+        }
+        else
+        {
+            _ = StartTimer();
+        }
+    }
+
+    private void UpdateTimer()
+    {
+        _timerSeconds--;
+
+        int seconds = _timerSeconds % 60;
+        int minutes = _timerSeconds / 60;
+
+        GetNode<Label>(TimerNodeName).Text = $"{minutes}:{seconds.ToString("00")}";
```

Comment:
There is a nice shorter formatting syntax when interpolating strings: `$"{expression:formatString}"`
```suggestion
        GetNode<Label>(TimerNodeName).Text = $"{minutes}:{seconds:00}";
```